### PR TITLE
Map small states #162

### PR DESCRIPTION
**Pull request by tmfrnz**

### CODE REVIEW

File: app/containers/MapContainer/utils.js [hunk 1]
```diff
@@ -57,6 +57,104 @@ export const valueOfCircle = (radius, range, config) => {
...
+  const events = {
+    mouseover: (e) => markerEvents.mouseover ? markerEvents.mouseover(e, config) : null,
+    mouseout: (e) => markerEvents.mouseout ? markerEvents.mouseout(e, config) : null,
+    click: (e) => (markerEvents.click ? markerEvents.click(e, config) : null),
+  };
+
+  const jsonLayer = L.geoJSON(data, {
+    pointToLayer: (feature, latlng) => {
+      const filterFeature = filterFeatureMaxZoom(feature, zoom);
+      const styles = getAreaLayer(feature, indicator, mapOptions, mapSubject, maxValueCountries, styleType, valueToStyle);
```
Comment (tmfrnz) [inline app/containers/MapContainer/utils.js:84]: @vbojilova ultimately I am not sure the marker styles will share anything with the area styles except for the fill color so not sure if a common style function is needed here

File: app/containers/MapContainer/utils.js [hunk 2]
```diff
@@ -57,6 +57,105 @@ export const valueOfCircle = (radius, range, config) => {
   return scale(radius);
 };
 
+export const filterFeatureMaxZoom = (feature, zoom) => {
```
Comment (tmfrnz) [inline app/containers/MapContainer/utils.js:60]: could be renamed to filterFeaturesByZoom

could be made more generic by passing the property name, i.e. `propertyMaxZoom` as an argument

File: app/containers/MapContainer/utils.js [hunk 3]
```diff
@@ -57,6 +57,105 @@ export const valueOfCircle = (radius, range, config) => {
...
+    mouseover: (e) => markerEvents.mouseover ? markerEvents.mouseover(e, config) : null,
+    mouseout: (e) => markerEvents.mouseout ? markerEvents.mouseout(e, config) : null,
+    click: (e) => (markerEvents.click ? markerEvents.click(e, config) : null),
+  };
+
+  const jsonLayer = L.geoJSON(data, {
+    pointToLayer: (feature, latlng) => {
+      const filterFeature = filterFeatureMaxZoom(feature, zoom);
+      const styles = getAreaLayer(feature, indicator, mapOptions, mapSubject, maxValueCountries, styleType, valueToStyle);
+      return L.circleMarker(latlng, {
```
Comment (tmfrnz) [inline app/containers/MapContainer/utils.js:85]: let's try be something like

```
const svgIcon = L.divIcon({
  html: `
<svg
  xmlns="http://www.w3.org/2000/svg"
  viewBox="0 0 28 30"
  width="28"
  height="30"
>
  <path d="m24,14.18c0-5.52-4.48-10-10-10S4,8.66,4,14.18c0,4.37,2.8,8.07,6.71,9.43l3.29,4.2,3.29-4.2c3.9-1.36,6.71-5.07,6.71-9.43Z" fill="${iconRingColor}"/>
  <circle cx="14" cy="14.18" r="8.18" fill="${iconCircleColor}"/>
</svg>`,
  className: "",
  iconSize: [28, 30], // may have to adjust this
  iconAnchor: [12, 40], // will have to adjust this
});
```

File: app/containers/MapContainer/utils.js [hunk 4]
```diff
@@ -57,6 +57,105 @@ export const valueOfCircle = (radius, range, config) => {
...
+    },
+  });
+
+  layer.addLayer(jsonLayer);
+
+  return layer;
+};
+
+// generate styles for a feature
+export const getAreaLayer = (feature, indicator, mapOptions, mapSubject, maxValueCountries, styleType, valueToStyle) => {
```
Comment (tmfrnz) [inline app/containers/MapContainer/utils.js:102]: if kept (probably not needed) should renamed to something like `getAreaLayerStyle` (as we are not returning a Leaflet layer object here)

File: app/containers/MapContainer/utils.js (hunk 3 above)
Comment (tmfrnz) [reply app/containers/MapContainer/utils.js:85]: of course we'd have to determine the fill color (or color of the inner circle) similar to the area style (depending on type of map) - the outer "ring" color should default to white and turn black if feature is active

**Comment by vbojilova** [comment]
- Moved countryData styling back into MapWrapper - [963847d](https://github.com/dumparkltd/advocacy-tracker-client/pull/165/commits/963847d7c52d77e2877e135f955807c479d5bbac)
- Changed the countryPoint circleMarkers to divIcons. They are coloured, the border goes from white to black onClick and there is a small shadow to separate them from the rest of the map. [d98bc26](https://github.com/dumparkltd/advocacy-tracker-client/pull/165/commits/d98bc26c7b6f46d9af00dfaf65244cfe4a36aef2)

**Pull request merged**